### ISSUE TITLE
What is the use of IndexedSet?

### ISSUE THREAD

**Issue by SergeStinckwich** [cormas/cormas#463]
What is the use of ``IndexedSet`` class in CORMAS? 

Apparently this is a combination of ``Set`` and ``OrderedCollection``, define as a subclass of ``OrderedCollection`` that redefine:

- ``-``
- ``addFirst:``
- ``addLast:``
- ``collect:``
- ``insert:before:``

Only used in 2 methods of ``CMSpaceModel`` but also mentioned in class comments like ``CMNetworkEntity`` for arcs and nodes instance variables. Might be because, when I migrate the initial version, I think I replace some ``IndexedSet`` by ``OrderedCollection``.

If we want to combine ``Set`` and ``OrderedCollection``, there is an existing implementation call ``OrderedSet`` that might be used instead (with more tests): https://github.com/pharo-containers/Containers-OrderedSet

**Comment by pierbo** [comment]
Indeed, IndexedSet is frequently used in Cormas: all the list of entities stored by CMCormasModel (theXXXs) return an IndexedSet.
If OrderedSet is more efficient, we can of course use it. But it should be able to answer any method from OrderedCollection.

**Comment by hernanmd** [comment]
When I used the OrderedCollection replacement of IndexedSet the image hang, so we should test also this in the context of the utilization of the app.
